nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -185,13 +187,14 @@ private static void printUsage() {
     }
 
     public static void main(String[] args) throws IOException, InterruptedException {
-        if (args.length < 1 || args.length > 3) {
+        if (args.length < 1 || args.length > 4) {
             printUsage();
             return;
         }
 
         File dumpFile = null;
         boolean verbose = false;
+        String statusHistoryDays = null;
```

Comment:
Using `java.time.Duration` to represent the parsed value would be a helpful approach.

Reply:
It is parsed on the server. Because RunNiFi::makeRequest provides a String argument to the BootstrapListener, the string value is required. I also added validation on the RunNiFi side.